File: NET/UniversityScheduleClient/UniversityScheduleException.cs
using System;
using System.Net;

namespace Mntone.UniversityScheduleClient
{
	/// <summary>
	/// UniversityScheduleException
	/// </summary>
	public sealed class UniversityScheduleException
		: Exception
	{

		internal UniversityScheduleException( UniversityScheduleExceptionReason reason )
			: base()
		{
			this._Reason = reason;
			this._ErrorCode = -1;
		}

		internal UniversityScheduleException( UniversityScheduleExceptionReason reason, Exception innerException )
			: base( string.Empty, innerException )
		{
			this._Reason = reason;
			this._ErrorCode = -1;
		}

		internal UniversityScheduleException( int errorCode )
		{
			this._Reason = UniversityScheduleExceptionReason.WEB_EXCEPTION;
			this._ErrorCode = errorCode;
		}

		/// <summary>
		/// Error reason
		/// </summary>
		public UniversityScheduleExceptionReason Reason { get { return this._Reason; } }
		private readonly UniversityScheduleExceptionReason _Reason;

		/// <summary>
		/// Error code (for WEB_EXCEPTION)
		/// </summary>
		public int ErrorCode { get { return this._ErrorCode; } }
		private readonly int _ErrorCode;
	}
}

File: JVM/univschedule-client/src/main/java/mntone/univschedule/client/UniversityScheduleException.java
package mntone.univschedule.client;

/**
 * UniversityClassScheduleException.
 */
public final class UniversityScheduleException extends RuntimeException
{
	private final UniversityScheduleExceptionReason mReason;
	private final int mErrorCode;

	/**
	 * Initialize a new university class schedule exception.
	 *
	 * @param reason the reason
	 */
	UniversityScheduleException( final UniversityScheduleExceptionReason reason )
	{
		this.mReason = reason;
		this.mErrorCode = -1;
	}

	/**
	 * Initialize a new university class schedule exception.
	 *
	 * @param reason         the reason
	 * @param innerException the inner exception
	 */
	UniversityScheduleException( final UniversityScheduleExceptionReason reason, final Exception innerException )
	{
		super( innerException );
		this.mReason = reason;
		this.mErrorCode = -1;
	}

	/**
	 * Initialize a new university class schedule exception.
	 *
	 * @param errorCode the error code
	 */
	UniversityScheduleException( final int errorCode )
	{
		this.mReason = UniversityScheduleExceptionReason.WEB_EXCEPTION;
		this.mErrorCode = errorCode;
	}

	/**
	 * Get reason.
	 *
	 * @return the reason
	 */
	public UniversityScheduleExceptionReason getReason()
	{
		return this.mReason;
	}

	/**
	 * Get error code.
	 *
	 * @return the error code
	 */
	public int getErrorCode()
	{
		return this.mErrorCode;
	}
}
